sympy/physics/quantum/tests/test_transforms.py

```diff
@@ -0,0 +1,75 @@
+"""Tests of transforms of quantum expressions for Mul and Pow."""
+
+from sympy.core.symbol import symbols
+from sympy.testing.pytest import raises
+
+from sympy.physics.quantum.operator import (
+    Operator, OuterProduct
+)
+from sympy.physics.quantum.state import Ket, Bra
+from sympy.physics.quantum.innerproduct import InnerProduct
+from sympy.physics.quantum.tensorproduct import TensorProduct
+
+
+k1 = Ket('k1')
+k2 = Ket('k2')
+k3 = Ket('k3')
+b1 = Bra('b1')
+b2 = Bra('b2')
+b3 = Bra('b3')
+A = Operator('A')
+B = Operator('B')
+C = Operator('C')
+x, y, z = symbols('x y z')
+
+
+def test_bra_ket():
+    assert b1*k1 == InnerProduct(b1, k1)
+    assert k1*b1 == OuterProduct(k1, b1)
+    # Test priority of inner product
+    assert OuterProduct(k1, b1)*k2 == InnerProduct(b1, k2)*k1
+    assert b1*OuterProduct(k1, b2) == InnerProduct(b1, k1)*b2
+
+
+def test_tensor_product():
+    # We are attempting to be rigourous and raise TypeError when a user tries
+    # to combine bras, kets, and operators in a manner that doesn't make sense.
+    # In particular, we are not trying to interpret regular ``*`` multiplication
+    # as a tensor product.
+    with raises(TypeError):
+        assert k1*k1 == TensorProduct(k1, k1)
```

Comment:
Which part of this is raising TypeError? Is it the `==` or the `k1*k2`. The with block should only have the minimal code that raises the error.

Reply:
Yeah, jusst the `k1*k2` part, will fix that now.